### DISCUSSION TITLE
Mirador 4 planning

### DISCUSSION THREAD

**Discussion by cbeer** [ProjectMirador/mirador#3457] [Ideas]
This is a start of a list of ideas + issues that may require a new major release of Mirador. 

## Technical architecture

React is moving (has moved?) away from `Component` classes towards functions and hooks. We might take this opportunity to align with upstream conventions:

- [ ] https://github.com/ProjectMirador/mirador/issues/2047 : Figure out how to test, ignore, or replace containers
- [ ] https://github.com/ProjectMirador/mirador/issues/1858 : Prefer Stateless Functions
- [ ] https://github.com/ProjectMirador/mirador/issues/2003 : Can we unbundle React?
- [ ] Consider alternative build frameworks (e.g. replacing webpack with vite) and test frameworks (e.g. replacing jest with vitest)
- [ ] Consider migrating projectmirador plugins to a mirador "monorepo" to ensure consistency across plugins and make core plugin development a little easier.

## UI/UX

Material UI 5.x is on the horizon. It's unclear if it will be a major breaking change, or something we can support in a minor release:
- [ ] https://github.com/ProjectMirador/mirador/issues/3221 : Test Material UI 5.alpha

## Plugin system

- [ ] https://github.com/ProjectMirador/mirador/issues/3160 "Componentless" plugins
- [ ] https://github.com/ProjectMirador/mirador/issues/2099 Make the plugin system easier to implement
- [ ] https://github.com/ProjectMirador/mirador/issues/2084 Provide non-redux "events"

## Revisit/review

These tickets have some ideas for potential work that wasn't pursued in Mirador 3 development.

https://github.com/ProjectMirador/mirador/issues/1693 : Define/revisit Mirador API Requirements
https://github.com/ProjectMirador/mirador/issues/2200 : Refactor the Mirador entry point 
https://github.com/ProjectMirador/mirador/issues/1991 : M3 non-functional requirements, concerns and issues
https://github.com/ProjectMirador/mirador/issues/2604 : Group components into folders
https://github.com/ProjectMirador/mirador/issues/1863 + https://github.com/ProjectMirador/mirador/issues/2951 : Naming conventions
https://github.com/ProjectMirador/mirador/issues/3258 : Validate used attributes
https://github.com/ProjectMirador/mirador/issues/2051 + https://github.com/ProjectMirador/mirador/issues/1628 : Redux payload format
https://github.com/ProjectMirador/mirador/issues/1808 : Generic fetching actions
https://github.com/ProjectMirador/mirador/issues/1749 : Improve (and make consistent) API error handling

**Comment by DavidHaskiya** [comment]
Hi,
Is there an update on when version 4 is planned for release?

The Scrum team I'm the product owner for at the Swedish National Archive  is currently investigating options for IIIF Viewers with a focus on search and display of OCR/HTR-texts similar to [Minutes of the Federal Council (1848-1972) - Search (admin.ch)](https://www.chgov.bar.admin.ch/search)

We're in doubt as to whether to focus our evaluation on the latest stable release, 3.3, or whether to work against the main branch with the hope it'll fairly soon become the 4.0-release.

Best regards,
David

Head of Unit, AI-lab and Data Services
Swedish National Archives

**Comment by brittnylapierre** [comment]
Hey everyone! I thought I'd write an update, in case anyone stumbles onto this issue.

Mirador 4 is in alpha and you can [find it on NPM](https://www.npmjs.com/package/mirador/v/4.0.0-alpha.2)

Developers in the Mirador Community are now using this issue: https://github.com/ProjectMirador/mirador/discussions/3933 to guide them towards a full release of Mirador 4
